SCORER.md

```diff
@@ -21,11 +25,231 @@ You may need to build your own scorer - your own _language model_ if:
 * You are training a speech recognition model for a particular domain - such as technical words, medical transcription, agricultural terms and so on
 * If you want to improve the accuracy of transcription
 
-DeepSpeech supports the _optional_ use of an external scorer - if you're not sure if you need to build your own scorer, stick with the built-in one to begin with.
+**DeepSpeech supports the _optional_ use of an external scorer - if you're not sure if you need to build your own scorer, stick with the built-in one to begin with**.
 
 ### Building your own scorer
 
-Building your own scorer is beyond the scope of the DeepSpeech Playbook, but the [DeepSpeech documentation covers how to do this](https://deepspeech.readthedocs.io/en/latest/Scorer.html). There are built-in scripts with DeepSpeech which make building an external scorer - a _language model_ - easier.
+_This section assumes that you are using a Docker image and container for training, as outlined in the [environment](ENVIRONMENT.md) section. If you are not using the Docker image, then some of the scripts such as `generate_lm.py` will not be available in your environment._
+
+_This section also assumes that you have already [trained a model](TRAINING.md) and have stored checkpoints for the trained model. This is required to generate the `--default_alpha` and `--default_beta` parameters that are used by the `lm_optimizer.py` script._
+
+The process for building your own scorer comprises the following steps:
+
+1. Having, or preparing, a text file (in `.txt` or `.txt.gz` format), with one phrase or word on each line. If you are training a speech recognition model for a particular domain - such as technical words, medical transcription, agricultural terms etc, then they would appear in the text file.
+
+2. Using the `lm_optimizer.py` script with _checkpoints_ from a trained model to generate values for the parameters `--default_alpha` and `--default_beta` that are used by the `generate_scorer_package` script.
+
+3. Using the `generate_lm.py` script which is distributed with DeepSpeech, along with the text file, to create two files, called `lm.binary` and `vocab-500000.txt`.
+
+4. Downloading the prebuilt `native_client` from the DeepSpeech repository on GitHub, and using the `generate_scorer_package` to create a `kenlm.scorer` file.
+
+5. Using the `kenlm.scorer` file as the _external_scorer_ passed to `DeepSpeech.py` during training.
+
+In the following example we will create a custom external scorer file for Bahasa Indonesia (BCP47: `id-ID`).
+
+#### Preparing the text file
+
+This is straightforward. In this example, we will use a file called `indonesian-sentences.txt`. This file has been created by copying phrases from the `train.tsv`, `validate.tsv` and `test.tsv` files in the Common Voice Indonesian dataset.
```

Comment:
While this is okay for the purpose of explaining how things work, this might be hinted as a "good way to do it" when actually you are biaising badly your system doing so.

Reply:
Given the target audience, I'd advertise, on the contrary, not to mix training/validation set as well as not test set data and suggest fetching data from elsewhere (a wikipedia text dump?)

Reply:
Excellent explanation, thank you.